components/job-orchestration/job_orchestration/scheduler/query/query_scheduler.py

```diff
@@ -67,9 +73,112 @@
 # Dictionary that maps IDs of file splits being extracted to IDs of jobs waiting for them
 active_file_split_ir_extractions: Dict[str, List[str]] = {}
 
+# Dictionary that maps IDs of clp-s archives being extracted to IDs of jobs waiting for them
+active_archive_json_extractions: Dict[str, List[str]] = {}
 reducer_connection_queue: Optional[asyncio.Queue] = None
 
 
+class StreamExtractionHandle(ABC):
+    def __init__(self, job_id: str):
+        self.job_id = job_id
+        self.archive_id: Optional[str] = None
```

Comment:
Can we make all the members private (in the base and derived classes) to enforce encapsulation?

Reply:
Done